### ISSUE TITLE
RCS9003 - Only works with binary expressions

### ISSUE THREAD

**Issue by glen-84** [dotnet/roslynator#1657]
It would be good if things like this were also flagged:

```cs
// 'configureServices' is not null here.
configureServices?.Invoke(services);
```

```cs
var pickle = input?.Pickles!.First()?.ButterPickle;
//                ^                 ^
```

etc.

**Comment by josefpihrt** [comment]
This analyzer is meant to be reported for special cases like IsKind which is extension method that actually allows first parameter to be null. It's meant to be used only when referencing Microsoft.CodeAnalysis* packages.

You are referring to a more general analysis if some expression can/cannot be null which is a different store. It may be possible to implement it, it also may turn out to be very tricky.
